pkg/controller/storage/mem/storage_broker.go

```diff
@@ -36,37 +36,58 @@ func newMemStorageBroker() *memStorageBroker {
 }
 
 func (m *memStorageBroker) List() ([]*servicecatalog.Broker, error) {
-	ret := make([]*servicecatalog.Broker, len(m.brokers))
+	copy := make([]*servicecatalog.Broker, len(m.brokers))
 	i := 0
-	for _, val := range m.brokers {
-		ret[i] = val
+	for _, br := range m.brokers {
+		copy[i] = &servicecatalog.Broker{}
+		if err := deepCopy(br, copy[i]); err != nil {
+			return nil, err
+		}
 		i++
 	}
-	return ret, nil
+	return copy, nil
 }
 
 func (m *memStorageBroker) Get(name string) (*servicecatalog.Broker, error) {
-	ret, ok := m.brokers[name]
+	br, ok := m.brokers[name]
 	if !ok {
 		return nil, errNoSuchBroker
 	}
-	return ret, nil
+	copy := &servicecatalog.Broker{}
+	if err := deepCopy(br, copy); err != nil {
+		return nil, err
+	}
+	return copy, nil
 }
 
 func (m *memStorageBroker) Create(br *servicecatalog.Broker) (*servicecatalog.Broker, error) {
 	if _, err := m.Get(br.Name); err == nil {
 		return nil, errBrokerAlreadyExists
 	}
-	m.brokers[br.Name] = br
-	return br, nil
+	m.brokers[br.Name] = &servicecatalog.Broker{}
+	if err := deepCopy(br, m.brokers[br.Name]); err != nil {
+		return nil, err
+	}
+	copy := &servicecatalog.Broker{}
+	if err := deepCopy(br, copy); err != nil {
+		return nil, err
+	}
+	return copy, nil
 }
 
 func (m *memStorageBroker) Update(br *servicecatalog.Broker) (*servicecatalog.Broker, error) {
 	if _, err := m.Get(br.Name); err != nil {
 		return nil, errNoSuchBroker
 	}
-	m.brokers[br.Name] = br
-	return br, nil
+	m.brokers[br.Name] = &servicecatalog.Broker{}
+	if err := deepCopy(br, m.brokers[br.Name]); err != nil {
+		return nil, err
```

Comment:
same comment here as in https://github.com/kubernetes-incubator/service-catalog/pull/261/files#r96036302, and the below `Update` implementations